clippy_lints/src/matches/collapsible_match.rs

```diff
@@ -108,6 +108,28 @@ fn check_arm<'tcx>(
                 "the outer pattern can be modified to include the inner pattern",
             );
         });
+    } else if let Some(inner) = IfLetOrMatch::parse(cx, inner_expr)
+        && let IfLetOrMatch::Match(inner_scrutinee, inner_arms, ..) = inner
+        && outer_pat.span.eq_ctxt(inner_scrutinee.span)
+        && let Some(inner_binding_id) = path_to_local(peel_ref_operators(cx, inner_scrutinee))
+        && let Some(outer_binding_id) = path_to_local(peel_ref_operators(cx, outer_then_body))
```

Comment:
Isn't this pattern here unreachable, considering that `outer_then_body` will be the match expression and never a path to a local variable? The test that was added already gets a warning today and the linked issue doesn't have a warning with this change, so it would be nice to see a test case for what this changes/one that didn't warn before

Reply:
I think it would be better if this lint rewrote the nested `match` into a simply simpler `match`... I will update this explanation with code examples later.